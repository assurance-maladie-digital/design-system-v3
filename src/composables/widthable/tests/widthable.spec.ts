import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { defineComponent } from 'vue'

import { Widthable } from '../'

function createTestComponent() {
	return defineComponent({
		mixins: [Widthable],
		template: '<div />',
	})
}

describe('Widthable', () => {
	it('computes the default styles', () => {
		const testComponent = createTestComponent()
		const wrapper = mount(testComponent)

		expect(wrapper.vm.widthStyles).toBe('width: 100%;')
	})

	it('computes the styles when min-width is defined', () => {
		const testComponent = createTestComponent()
		const wrapper = mount(testComponent, {
			propsData: {
				minWidth: '512px',
			},
		})

		expect(wrapper.vm.widthStyles).toBe('min-width: 512px; width: 100%;')
	})

	it('computes the styles when max-width is defined', () => {
		const testComponent = createTestComponent()
		const wrapper = mount(testComponent, {
			propsData: {
				maxWidth: '512px',
			},
		})

		expect(wrapper.vm.widthStyles).toBe('max-width: 512px; width: 100%;')
	})
})
