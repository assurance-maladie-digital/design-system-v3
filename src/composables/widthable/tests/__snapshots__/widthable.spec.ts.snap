// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Widthable > computes the default styles 1`] = `undefined`;

exports[`Widthable > computes the styles when max-width is defined 1`] = `undefined`;

exports[`Widthable > computes the styles when min-width is defined 1`] = `undefined`;
